source/index.ts

```diff
@@ -206,7 +215,18 @@ export default class PQueue<QueueType extends Queue<EnqueueOptionsType> = Priori
 				this._intervalCount++;
 
 				try {
-					resolve(await fn());
+					const operation = this._timeout === undefined ? fn() : pTimeout(
+						Promise.resolve(fn()),
+						this._timeout,
+						() => {
+							if (this._throwOnTimeout) {
+								reject(timeoutError);
+							}
+
+							return undefined;
```

Comment:
This is useless

```suggestion
```

Reply:
Nope, implicit `return undefined` aka void is different in its type signature in this case in terms of `<void>` vs `<undefined>` ✌️

Reply:
`return undefined` is not `void`. You haven't still explained why you need `return undefined`.

Reply:
exactly! and even though we'd like to just `void` here, the compiler won't let us: 
<img width="981" alt="Screenshot 2019-04-27 at 16 27 11" src="https://user-images.githubusercontent.com/8214559/56850996-361d7380-690a-11e9-984d-3d42bd8ef6e2.png">

the resolve function anyhow infers that `undefined` would be okay 
<img width="988" alt="Screenshot 2019-04-27 at 16 33 58" src="https://user-images.githubusercontent.com/8214559/56851002-4f262480-690a-11e9-89a6-a52f62b36c34.png"> so I went with that instead.

So we have to return just anything in case the promise times out and we don't throw/reject. Would you prefer resolving with `timeoutError` thus changing the function signature?

Reply:
Thanks, didn't know about that :O I appreciate it.